lib/pub_sub/subscriber.rb

```diff
@@ -66,5 +66,19 @@ def queue_arn(region)
     def queue_url(region)
       sqs(region).create_queue(queue_name: PubSub.service_identifier).queue_url
     end
+
+    # Collects all :topics or :subscriptions for a given SNS client
+    def self.collect_all(client, collection_type)
```

Comment:
Did you run into any issues with an approach like this? As instance methods like #topics(*regions) and #subscriptions(*regions) since your doing the region iteration repeatedly in the tasks. 

```ruby

subscriber = PubSub::Subscriber.new
subscriber.sns_subscriptions.each do |subscription|
  # work 
end

subscriber.sns_topics.each do |topic|
  # work 
end
```